### PR TITLE
[TC Automation:PTX-15465] HAIncreasePoolresizeAndAdddisk 

### PR DESCRIPTION
**Pull request by spai-px**
What this PR does / why we need it:
Adds test case HAIncreasePoolresizeAndAdddisk . 

Which issue(s) this PR fixes:
Closes # PTX-15465
[Jira Ticket](https://portworx.atlassian.net/browse/PTX-15465)

Special notes for your reviewer:
Test run: [Aetos](https://aetos.pwx.purestorage.com/resultSet/testSetID/548267  )
[jenkins](https://jenkins.pwx.dev.purestorage.com/job/Users/job/krishna/job/tp-pxe-byoc/1579/console)

### CODE REVIEW

File: tests/basic/storage_pool_test.go [hunk 1]
```diff
@@ -10626,3 +10626,154 @@ var _ = Describe("{PoolDeleteServiceDisruption}", func() {
...
+					newRepl := int64(curReplSet - 1)
+					log.FailOnError(Inst().V.SetReplicationFactor(vol, newRepl,
+						nil, nil, true),
+						"Failed to set Replicaiton factor")
+				}
+				// now increase the replication factor of this volumes.
+				nodeToBeUpdated = node.GetStorageDriverNodes()[rand.Intn(len(node.GetStorageDriverNodes()))]
+				pools, err := GetAllPoolsOnNode(nodeToBeUpdated.Id)
+				poolToBeUpdated = pools[0]
+				log.FailOnError(err, "Failed to get all pools on node: %v", nodeToBeUpdated.Name)
```
Comment (dbhatnagar-px) [inline tests/basic/storage_pool_test.go:10688]: Move this line right after GetAllPoolsOnNode

File: tests/basic/storage_pool_test.go [hunk 2]
```diff
@@ -10626,3 +10626,154 @@ var _ = Describe("{PoolDeleteServiceDisruption}", func() {
...
+				log.InfoD("Node selected for repl increase")
+				var maxReplicaFactor int64
+				var nodesToBeUpdated []string
+				var poolsToBeUpdated []string
+				nodesToBeUpdated = append(nodesToBeUpdated, nodeToBeUpdated.Name)
+				poolsToBeUpdated = append(poolsToBeUpdated, poolToBeUpdated)
+				maxReplicaFactor = 3
+				nodesToBeUpdated = nil
+				poolsToBeUpdated = nil
```
Comment (dbhatnagar-px) [inline tests/basic/storage_pool_test.go:10697]: Pls remove these lines

File: tests/basic/storage_pool_test.go [hunk 3]
```diff
@@ -10626,3 +10626,154 @@ var _ = Describe("{PoolDeleteServiceDisruption}", func() {
...
+				log.FailOnError(Inst().V.SetReplicationFactor(vol, maxReplicaFactor,
+					nodesToBeUpdated, poolsToBeUpdated, true),
+					"Failed to set Replicaiton factor")
+				// create a time limit for this for loop
+				timeout := time.After(20 * time.Minute)
+				for {
+					volDetails, err := Inst().V.InspectVolume(vol.ID)
+					log.FailOnError(err, "Failed to get volume details")
```
Comment (dbhatnagar-px) [inline tests/basic/storage_pool_test.go:10705]: Please use task.DoRetryTimeout

**Review by dbhatnagar-px — CHANGES_REQUESTED** [review]

File: tests/basic/storage_pool_test.go (hunk 1 above)
Comment (spai-px) [reply tests/basic/storage_pool_test.go:10688]: Done

File: tests/basic/storage_pool_test.go (hunk 2 above)
Comment (spai-px) [reply tests/basic/storage_pool_test.go]: Done

File: tests/basic/storage_pool_test.go (hunk 3 above)
Comment (spai-px) [reply tests/basic/storage_pool_test.go]: Done

**Review by dbhatnagar-px — APPROVED** [review]
Lgtm

**Pull request merged**